messaging/src/main/java/org/axonframework/messaging/timeout/UnitOfWorkTimeoutInterceptor.java

```diff
@@ -0,0 +1,105 @@
+     * on the {@link AxonTaskJanitor#INSTANCE}. If you want to use a different {@link ScheduledExecutorService} or
+     * {@link Logger} to log on, use the other
+     * {@link #UnitOfWorkTimeoutInterceptor(String, int, int, int, ScheduledExecutorService, Logger)}.
+     *
+     * @param componentName    The name of the component to be included in the logging
+     * @param timeout          The timeout in milliseconds
+     * @param warningThreshold The threshold in milliseconds after which a warning is logged. Setting this to a value
+     *                         higher than {@code timeout} will disable warnings.
+     * @param warningInterval  The interval in milliseconds between warnings.
+     */
+    public UnitOfWorkTimeoutInterceptor(String componentName,
+                                        int timeout,
+                                        int warningThreshold,
+                                        int warningInterval
+    ) {
+        this(componentName,
+             timeout,
+             warningThreshold,
+             warningInterval,
+             AxonTaskJanitor.INSTANCE,
+             AxonTaskJanitor.LOGGER);
+    }
+
+    /**
+     * Creates a new {@link UnitOfWorkTimeoutInterceptor} for the given {@code componentName} with the given
+     * {@code timeout}, {@code warningThreshold} and {@code warningInterval}. The warnings and timeout will be scheduled
+     * on the provided {@code executorService}.
+     *
+     * @param componentName    The name of the component to be included in the logging
+     * @param timeout          The timeout in milliseconds
+     * @param warningThreshold The threshold in milliseconds after which a warning is logged. Setting this to a value
+     *                         higher than {@code timeout} will disable warnings.
+     * @param warningInterval  The interval in milliseconds between warnings.
+     * @param executorService  The executor service to schedule the timeout and warnings
+     * @param logger           The logger to log warnings and errors
+     */
+    public UnitOfWorkTimeoutInterceptor(String componentName,
+                                        int timeout,
+                                        int warningThreshold,
+                                        int warningInterval,
+                                        ScheduledExecutorService executorService,
+                                        Logger logger
+    ) {
+        this.componentName = componentName;
+        this.timeout = timeout;
+        this.warningThreshold = warningThreshold;
+        this.warningInterval = warningInterval;
+        this.executorService = executorService;
+        this.logger = logger;
+    }
+
+    @Override
+    public Object handle(@Nonnull UnitOfWork<? extends Message<?>> unitOfWork,
+                         @Nonnull InterceptorChain interceptorChain) throws Exception {
+        if (!unitOfWork.resources().containsKey("_transactionTimeLimit")) {
```

Comment:
Were is the `_transactionTimeLimit` resource key set?

Reply:
Nowhere! :exploding_head: Good catch